src/Compress.jl

```diff
@@ -63,45 +63,43 @@ function load(path::AbstractString,
                 if haskey(uv, key)
                     error("Overlapping ranges for $(key) in Compat. Detected for version $(v).")
                 else
-                    uv[key] = value
+                    # The value can be either a UUID or a VersionSpec
+                    value_T = tryparse(Base.UUID, value)
+                    if value_T === nothing
+                        value_T = VersionSpec(value)
+                    end
+                    uv[key] = value_T
                 end
             end
         end
     end
     return uncompressed
 end
 
-function compress(path::AbstractString, uncompressed::Dict,
+function compress(path::AbstractString, uncompressed::Dict{VersionNumber,Dict{String,Union{Base.UUID, VersionSpec}}},
                   versions::Vector{VersionNumber} = load_versions(path))
-    inverted = Dict{Pair{String,Any},Vector{VersionNumber}}()
+    inverted = Dict{Pair{String,Union{Base.UUID, VersionSpec}},Vector{VersionNumber}}()
     for (ver, data) in uncompressed, (key, val) in data
-        val isa Base.UUID && (val = string(val))
```

Comment:
to check my understanding, this is the main change? we now match before stringifying rather than after so we use the actual semantics instead of string equality

Reply:
Yes